### DISCUSSION TITLE
PRODUCT_DOCUMENTATION npl_class: Parsing and Consolidation ?

### DISCUSSION THREAD

**Discussion by cverluise** [cverluise/PatCit#26] [Projects & Ideas]
Hello,

Since `v0.2-npl`, we have introduced a PRODUCT_DOCUMENTATION `npl_class`

### Question
How should this class of citations be structured and consolidated?

### Food for thoughts
Below is a sample of 100 citations for this class.

<details>

```sql
WITH
  tmp AS (
  SELECT
    npl_publn_id
  FROM
    `npl-parsing.patcit.v02_npl`
  WHERE
    npl_class="PRODUCT_DOCUMENTATION")
SELECT
  npl_biblio
FROM
  `usptobias.patstat.tls214` AS tls214,
  tmp
WHERE
  tmp.npl_publn_id = tls214.npl_publn_id
LIMIT 100  
```

Row | npl_biblio |  
-- | -- | --
1 | Teva Pharmaceuticals USA, Desmopressin Acetate Tablet Product Description, (Feb. 2008). |  
2 | Wall Street Journal, Aug. 13, 1998, B1, Col. 2. |  
3 | Essex Specialty Products, Inc., Betamate Structural Adhesives Testing Procedures For Prepainted Metal , 22 pages. |  
4 | Hitachi Product Literature 1982 Tri clad metal for button type batteries, (No Month). |  
5 | Microsoft Corporation, Programmer s Guide to Microsoft Windows 95: Key Topics on Programming for Windows from the Microsoft Windows Development Team, 130, 1995. |  
6 | Leatherman Super Tool Product Description, issued before Oct. 7, 1996. |  
7 | 10.4 Touchscreen LCD, PixelTouch Brochure. |  
8 | Turbo Cove Groover, User Guide Feb. 20, 2007. |  
9 | 'Graphics: Intel® 82852/828b5 Graphics Controller Family', Intel, Archived Nov. 2, 2006 by archive.org, Downloaded Jun. 30, 2011, http://web.archive.org/web/20061103045644/http://www.intel.com/support/graphics/intel852gm/sb/CS-009064.htm? (Per 37 C.F.R. 1.98(d), this reference is not required and is not submitted because this reference was previously cited by or submitted to the PTO in prior U.S. Appl. No. 10/390,120 to which the present application. |  
10 | NET-SAFE, downloaded from http://www.acmepacket.com/html/page.asp?PageID=%7BFB2657BA-EE7A-46C1-BEA8-, on May 16, 2008 , 4 pages. |  
11 | Product Literature, 'The MIT Wearable Computing Page', [retrieved on May 8, 1997]. Retrieved from the Internet: <URL: http://lcs.www.media.mit.edu>. |  
12 | Google Archive search, demonstrating search result for MAX 5073 data sheet online on Jan 11, 2005 |  
13 | Adaptive Optics Assoc., Inc., Capture motion, release creativity, 54 Cambridge Park Dr., Cambridge, MA 02140 2308, Product brochure MT202 CM, Aug. 1994. |  
14 | 'FIELDVUE(R) VL2000 Series Software User Guide,' Fisher Controls, Version 2.0 (Jun. 1996). |  
15 | Material Safety Data Sheet, EKC Technology, Inc., Aug. 1990. |  
16 | Phaup, 'Question Mark Introduces Access Export Software,' Web page, unverified print date of Apr. 2, 2002, unverified cover date of Mar. 1, 1997. |  
17 | Ribbit.com, “What is Ribbit? Features,” http://www.ribbit.com/platform/features.php, Downloaded on Sep. 26, 2008, 1 page. |  
18 | Manitoba Agriculture, Food and Rural Initiatives www.gov.mb.ca/agriculture/crops/hemp/bko05s12.html, Fibre Production, 2 Pages, No Date. |  
19 | Two page two-sided color brochure entitled 'Drying Systems: WD Series Dehumidifying Dryers' of AEC Whitlock, 1997. |  
20 | David, “Poker Academy Pro 2.0”, Gamezone, Review Date—Oct. 24, 2005, retrieved from Internet Archive on 1/222/2015 from Internet URL<http://pc.gamezone.com/gzreviews/r27947.htm>. |  
21 | Huff, KL, “Data Set Usage Sequence Number,” IBM Technical Disclosure Bulletin, vol. 24, No. 5, Oct. 1981 New York, US, pp. 2404-2406. |  
22 | Brochure by Neles Jamesbury entitled Neldisc Metal Seated Butterfly Valves , Issue Apr. 1989, pp. 1 2. |  
23 | Tellurex Corp. 'Thermoelectric cooling semiconductor modules available in new configuration' MPMN: Cover Products Apr. 1999 http://www.devicelink.com/mpmn/archive/99/04/cover.html. |  
24 | The MathWorks, 'Simulink Fixed Point, for Use with Simulink, User's Guide, Version 5,' (2005). |  
25 | i960 CA Microprocessor User s Manual, Intel Corporation, 1992, pp. iii, 2 1 through 2 4. |  
26 | HSNW: SRI defense technology spawns civilian application: published Jun. 29, 2010. |  
27 | Hewlett Packard, PA-RISC 1.1 Architecture and Instruction Set Reference Manual, 3rd ed. Feb. 1994, pp. 1-424. |  
28 | Smith Berger Innovation: Deep Water Mooring Systems (Offshore Incorporating The Oilman, Apr. 1989) p. 18. |  
29 | Aube T1062 User Manual, published Jul. 4, 2008. |  
30 | 6DN5MM4K: '马铃薯高产栽培技术要点', 《豆丁网，HTTP://WWW.DOCIN.COM/P-780808991.HTML》 |  
31 | Quanta Biodesign, Ltd., Leading innovator, producer and provider of monodisperse discrete Pegtm (dPEG(TM)) derivatives, (Product Catalog), 26 pages, (Updated: Nov. 17, 2005). |  
32 | Intel Pentium Processor with MMX Technology , Jun. 1997 243185.004. |  
33 | Sharfi, Benjamin; 'VMEbus advances aim at high data rates'; General Micro Systems Inc.; May 8, 2000; accessed via Internet 12/23/2013 . |  
34 | Electronic Products , Patented contact yields high interconnection density , May 1996. |  
35 | SAMBROOK; RUSSELL: 'Molecular Cloning: A Laboratory Manual', 2000, COLD SPRING HARBOR PRESS |  
36 | Note Runner Advertisement, written by Extreme Gaming, published prior to Sep. 30, 2004. |  
37 | Transport Provider Interface Specification, Published by UNIX International, Waterview Corporate Center, 20 Waterview Boulevard, Parsippany, NJ 07054, 1992. |  
38 | 'Modular Object Scanning Technology', Version 1.10, User Manual, FabSoft, Inc., 2006, pp. 1-26. |  
39 | Habraken, Microsoft Office XP 8 in 1, published Jun. 5, 2001, http://proquest.safaribooksonline.com/0789725096/, 12 pp. |  
40 | Brochure, Roadtec SP 100 Gravity Feed Paver (undated). |  
41 | Headend Combining/Splitting Network , Cox Communications, (undated), 16 pgs. |  
42 | Weiss et al., User s Manual for the Basic Math Mastery Tests , 1992, Abstract. |  
43 | Cerulean Studios, 'Trillian Online User Manual,' http://www.ceruleanstudios.com/support/manual.php?hchap=4&hsub=l&hsect=5, 2008. |  
44 | 1997 Nordx/CDT catalog entitled 'IBDN Catalog and Reference Guide,' front cover page, pp. 5-21 through 5-29, and back cover page, Oct. 1996. |  
45 | “Microsoft Office 2003 Editions Product Guide”, Published on: Sep. 2003, Available at: tp://www.google.ca/url?sa=t&rct=j&q=&esrc=s&frm=1&source=web&cd=3&ved=0CCoQFjAC&url=http%3A%2F%2Fdownload.microsoft.com%2Fdownload%2F0%2Ff%2F1%2F1%2F0f1d5b1f-53bc-47c3-bf6f-ac6d67cf9766%2FOffice2003Guide.doc&ei=CKTkVPBCgb-xBL6qgvAH&usg=AFQjCNEpNk4IFH6m27BXr48IKv1kRXo—xA, 167 pgs. |  
46 | 21164 Alpha Microprocessor Data Sheet, Cover, Table of Contents, Chapters 6, 8 and 10, Samsung Electronics, c. 1997. |  
47 | Lecture Collections in 96 Ferroelectric Film Memory Technique Forum, Science Forum, Inc., p. 4 4, lines 1 12, Ishihara, Jan. 26, 1996. |  
48 | 'DB2 Universal database XML Extender: Web-Enabling Your Data with XML', IBM product information sheet 2000, 2 pages. |  
49 | 'TMS320C6678 Multicore Fixed and Floating-Point Digital Signal Processor', Data Manual, SPRS691, Texas Instruments Incorporated, Nov. 2010, pp. 1-259. |  
50 | Plysolene PIB product sheet; 2013. |  
51 | User's Manual, Digital Writing System 2004, Logitech io2, 12 pages. |  
52 | LookSmart s Beseen; http://www.beseen.com, 1999. |  
53 | Brochure, Auto Restoration, GE Harris Control Systems Canada, Inc., 1997 (No month). |  
54 | Bard: Pelvic Organ Prolapse 'An overview of Pelvic Organ Prolapse' Brochure, C.R. Bard, Inc. Covington, GA 30014, www.bardmedical.com © 2011 C.R. Bard, Inc. |  
55 | '2X Sigma Spa® Brush Cleaning Glove.' Retrieved Dec. 14, 2015 from <https://web.archive.org/web/20140529021235/http://www.sigmabeauty.com/2X-Sigma-Spa-Brush-Cleaning-glove-p/ssg2x.htm>. |  
56 | Catalog-SoloEtte Steel String-Travel Practice Guitar-Publication date unknown. |  
57 | Cash Secure Internet Payment Service CyberCash s Secure Internet Payment Services , CyberCash, Inc., Reston, Virginia 22091. |  
58 | Gesco Pla Nate Palate Guard Brochure, GESCO International, Inc., a subsidiary of MedChem Products, Inc., Mar. 1993. |  
59 | Alignment Procedure for the PixSys Two Emitter Offset Probe for the SAC GP 8 3d Sonic Digitizer (undated) by PixSys, Inc., 3 unnumbered pages. |  
60 | Brochure from ADC Telecommunications entitled 'Fiber Optic Couplers, Splitters and Wavelength Division Multiplexers,' 27 pages, dated Jul. 1997. |  
61 | The Near Indestructible, Maintenance Free Rubber Door Systems Brochure, M & I Door Systems Limited, Barrie, Ontario, Canada, 2 pages. |  
62 | H-Pette(R) Dispenser, Helena Laboratories Website (brochure date unknown, but trademark registered showing first use in commerce in Nov. of 1993 as shown in attached registration printout from PTO)-Total pp.: 6. |  
63 | ISOCHEM product sheet for 5-butyl-5-ethyl-1,3-Dioxan-2-one, (2016). |  
64 | Rick Broida, The Four Best Phone Stands Money Can't Buy, Mar. 22, 2010, retrieved from: http://howto.cnet.com/8301-11310-39-10469907-285/the-four-best-phone-stands-money-cant-buy/. |  
65 | Vortex Brochure, JVL Corporation, 2005, 2 pages. |  
66 | Lark/Larkpro, User Manual, (2012) 'What's in the box,' Lark Technologies, 7 pp. |  
67 | Cheertainer image from CDF Corporation's web site (http://www.cdf1.com/images/banner—image/banner—cheertainer1.jpg), printed Jul. 15, 2009 (1 page). |  
68 | DuPont Engineering Plymers Product Information, Electrical/Electronic Thermoplastic Encapsulation . |  
69 | Born Free, Product insert brochure-shows Born Free Training and Drinking Cups exploded, cleaning, assembly and disassembly instructions. Color-1 page. Copyright Jul. 2008. |  
70 | Pentium Pro Family Developer s Manual, vol. 1: Specifications, 1996, Chapters 7, Cache Protocol. |  
71 | SAMBROOK; RUSSEL: 'Molecular Cloning: A Laboratory Manual(3rd edition),', 2001, COLD SPRING HARBOR LABORATORY PRESS |  
72 | Igo.com product web page printout for 'NBL-100 Notebook Light' by Interex; http://www.igo.com/cgi-bin/ncommerce3/ProductDisplay?prmenbr=1&prrfnbr-522530;printed Jul. 21,2000. |  
73 | Sigmacote® Brochure, Sigma Chemical Company, P.O. Box 14508, St. Louis, MO 63178, 3 pages, Apr. 28, 1997. |  
74 | Press Release, 'WMS Gaming's Monopoly Slot Machines Named 1998's Most Innovative Gaming Product At the American Gaming, Lodging and Leisure Summit,' published by WMS Gaming Inc. on Jan. 13, 1999. |  
75 | 赵芳: '一种医学图像体绘制的改进方法', 《中国优秀硕士学位论文全文数据库_信息科技辑》 |  
76 | Covidien SILS port, www.covidien.com/silsport/pages.aspx, 1 pg., website last visited Nov. 21, 2012. |  
77 | Extreme CCTV Inc., 'Extreme CCTV Releases ALERT Software for Remote License Plate Readers', Jan. 23, 2004. |  
78 | 'W3C Scalable Vector Graphics - What's new' (www.w3.org) 11-02-1999. Retrieved on the 19-04-2004 via: http://web.archive.org/web/19990508122454/http://www.w3.org/Graphics/SVG/ |  
79 | 'First in Thirst,' MatAsorb Industrial Sorbents product brochure (sourch and date unknown). |  
80 | IMechE Autotech 95 paper C498/38/234 presented 9/11/95 |  
81 | Creative: 'Creative NOMAD MuVo TX,' www.creative.com, Nov. 1, 2004, http://web.archive.org/web/20041024175952/www.creative.com/products/pfriendly.asp?product=9672 [downloaded Jun. 6, 2006]. |  
82 | SAMBROOK; RUSSELL: 'Molecular Cloning: A Laboratory Manual', 2001, COLD SPRING HARBOR LABORATORY PRESS |  
83 | Microsoft Works: Ten Minutes to Productivity, Microsoft Corporation, 1989. |  
84 | 'Peter Beebee's Home Page,' web page print-out (Feb. 12, 1999) from http://www-swiss.ai.mit.edu/htbin/ptbbgate/jwz/?fetch+personal%2Fmain.text.html (2 pages). |  
85 | Datasheet, Allegro Microsystems, Inc., 'ATS637LSA, True Power on, Self-Calibrating, Zero Speed Gear Tooth Sensor System,' Aug. 2001, 13 pages. |  
86 | Microsoft Windows v3.11, Microsoft Corp., screen dumps 1 6, 1993. |  
87 | Ford Motor Company, 'SYNC,' Owner's Guide Supplement, SYNC System Version 2 (Oct. 2008). |  
88 | Software Guard Extensions Programming Reference, 2010-2013, Intel Corporation. |  
89 | Cox, Product Catalog, 2004 (16 pages). |  
90 | NXP Product Datasheet for part SSL1750 titled 'SMPS control IC for LED drivers,' pp. 1-28, Rev. 01, Sep. 15, 2008. |  
91 | Geophysical Survey Systems Inc., SIR System 2 Subsurface Interface Radar, Brochure. |  
92 | 'Local Lisa® Intracardiac Navigation System Model 9670000/9670025.' Technical Manual Version 1.2, Chapter 1, pp. 1-19. 2004. |  
93 | Skeen, Dale Ph.D. and Vitria Technology, Inc. Velociti The Enterprise Capable Publish Subscribe Server, located at Internet World Wide Web Site http://www.vitria.com/whitepapers/velocitiwp.html, printed pp. 1 17, date printed from Internet Jan. 14, 1998, actual date of paper unknown. |  
94 | 'CY8C21x34 Data Sheet,' CSR User Module (CSR v1.0), Cypress Semiconductor Corp., Oct. 6, 2005, pp. 1-36. |  
95 | Mosby's Medical, Nursing & Allied Health Dictionary, Sixth Edition, p. 1638 (2002). |  
96 | Oracle Unveils Oracle Websystem Internet Line , Computergram International, Nov. 1995. |  
97 | Magellan Maestro 3100 User Manual, 2007, Magellan Navigation Inc, 631459-01 A, pages Cover thru iii, 11 and 12 |  
98 | 'Orthopaedic Product News', Aug. 2005, Retrieved from the Internet: URL:http://www.orthoworld.com/us-opn-2005-08.pdf [retrieved on May 26, 2009], p. 30, Hallux Valgus Correction with a Low Profile Locking Plate. |  
99 | Believe it or Not Article written by Strictly Slots, published in 2001. |  
100 | Battenfeld Technologies, Inc., 'Steady Rest Portable Shooting Rest,' 1 page [Internet accessed Jan. 25, 2006].

</details>

### Open questions

- Need for further sub-classification? If yes, which sub-classes?
- Fields to be parsed?
- Corpus/ reference database for consolidation?

I leave these questions open to discussion.

Cheers

**Comment by gderasse** [comment]
Here is a possible structure: **Company**; **productName**; **documentType**; **documentReference**

With **documentType** taking values such as {'Webpage', 'Product catalogue', 'User guide', 'Data sheet', 'Technical manual'}

**Comment by cverluise** [comment]
For parsing, I would add the url + date (if any) to be able to reconstruct the wayback engine url

For consolidation, some ideas
- google knowledge graph (cat {Thing, Product}): gtin_x,_ description
- aws product api
- 1st url of query "Comapny"+"productName" (Bing or sthg else)

**Comment by gderasse** [comment]
I particularly like the knowledge graph idea, though I wasn't able to find much information on the topic. Doesn't it include AWS products by the way?

**Comment by cverluise** [comment]
The Google knowledge graph was built on top of [Freebase](https://en.wikipedia.org/wiki/Freebase). I guess that it also includes wikidata entities. It does not seem to include AWS products.

The `Product` entity type schema (from the Google Knowledge graph) is described here https://schema.org/Product. 

Note: I performed some tests and found that the Google Knowledge graph was not fully satisfying. Also, the cat `Product` and `Thing` do not seem to be clearly  distinguished.
